File: app/src/main/java/com/nelyan/ui/TraderActivity.java
package com.nelyan.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.TimePickerDialog;
import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.TimePicker;

import com.nelyan.HELPER.image;
import com.nelyan.R;
import com.nelyan.adapter.DayTimeRepeatAdapter;
import com.nelyan.adapter.DescriptionRepeatAdapter;
import com.nelyan.adapter.EventRepeatAdapter;
import com.nelyan.adapter.TimeRepeatAdapter;
import com.nelyan.modals.DayTimeModel;
import com.nelyan.modals.TimeModel;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.HashMap;
import java.util.List;

import ru.tinkoff.scrollingpagerindicator.ScrollingPagerIndicator;

public class TraderActivity extends image implements
        AdapterView.OnItemSelectedListener {
    Context mContext;
    ImageView ivBack, ivImg, ivCam;
    Button btnSubmit;
    TextView tvAdd, tvClock;
    TextView edClo, edClo1, edClo2, edClo3;
    ImageView ivplus,ivImg1,ivImg2,ivImg3;
    LinearLayout ll_1, ll_2,ll_3;
    Spinner orderby, orderby1;    RecyclerView Recycler_scroll;
    ScrollingPagerIndicator indicator;
RelativeLayout rlAddImg,rlImg;
    RecyclerView rvDayTime,rvDesc;
    DayTimeRepeatAdapter dayTimeRepeatAdapter;
    static String imgtype,imasgezpos;
    int returnItemView=1;
    HashMap<String, Bitmap> image=new HashMap<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_trader);
        mContext = this;
       // indicator=findViewById(R.id.indicator);
       /* Recycler_scroll=findViewById(R.id.Recycler_scroll);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getApplicationContext(), LinearLayoutManager.HORIZONTAL, false);
        Recycler_scroll.setLayoutManager(linearLayoutManager);
        ItemsAdapter  adapterItems = new ItemsAdapter(mContext);
        Recycler_scroll.setAdapter(adapterItems);
        indicator.attachToRecyclerView(Recycler_scroll);*/
        rvDesc = findViewById(R.id.rvDesc);
        rvDayTime = findViewById(R.id.rvDayTime);

        orderby = findViewById(R.id.orderby);
        List<String> country = new ArrayList<>();
        country.add("");
        country.add("USA");
        country.add("Japan");
        country.add("India");
        ;
        ArrayAdapter arrayAdapte1 = new ArrayAdapter(this, R.layout.customspinner, country);
        orderby.setAdapter(arrayAdapte1);

        ivBack = findViewById(R.id.ivBack);
        btnSubmit = findViewById(R.id.btnSubmit);
        ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        btnSubmit = findViewById(R.id.btnSubmit);
        btnSubmit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent i = new Intent(TraderActivity.this, TraderPreviewActivity.class);
                //  i.putExtra("nurActivity","nurFrag");
                startActivity(i);
            }
        });
        ll_1 = findViewById(R.id.ll_1);
        ll_2 = findViewById(R.id.ll_2);
        ll_3 = findViewById(R.id.ll_3);

        ivImg = findViewById(R.id.ivImg);

        ivImg1 = findViewById(R.id.ivImg1);
        ivImg2 = findViewById(R.id.ivImg2);
        ivImg3 = findViewById(R.id.ivImg3);

        rlImg = findViewById(R.id.rlImg);
        rlImg.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                imgtype="0";
                image("all");
            }
        });ivplus = findViewById(R.id.ivplus); rlAddImg = findViewById(R.id.rlAddImg);
        rlAddImg.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                imgtype="1";
                image("all");
            }
        });  ll_1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                imgtype="2";
                image("all");
            }
        });ll_2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                imgtype="3";
                image("all");
            }
        });ll_3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                imgtype="4";
                image("all");
            }
        });

        final ArrayList<TimeModel> arrayList = new ArrayList();
        TimeModel timeModel = new TimeModel();
        timeModel.setEndtime("");
        timeModel.setStarttime("");
        arrayList.add(timeModel);

        final DayTimeModel dayTimeModel = new DayTimeModel(arrayList);

        final ArrayList<DayTimeModel> dayTimeModelArrayList = new ArrayList<>();
        dayTimeModelArrayList.add(dayTimeModel);

        dayTimeRepeatAdapter = new DayTimeRepeatAdapter(this, dayTimeModelArrayList,
                new DayTimeRepeatAdapter.DayTimeRepeatListener() {
                    @Override
                    public void dayTimeAdd(int pos) {
                        final ArrayList<TimeModel> arrayList1 = new ArrayList();

                        TimeModel timeModel = new TimeModel();
                        timeModel.setEndtime("");
                        timeModel.setStarttime("");
                        arrayList1.add(timeModel);
                        DayTimeModel dayTimeModel1 = new DayTimeModel(arrayList1);
                        dayTimeModelArrayList.add(dayTimeModel1);
                        dayTimeRepeatAdapter.notifyDataSetChanged();
                    }

                    @Override
                    public void timeAdd(int pos) {

                          DayTimeModel dayTimeModel1 = dayTimeModelArrayList.get(pos);
                        TimeModel timeModel = new TimeModel();
                        timeModel.setEndtime("");
                        timeModel.setStarttime("");

                        dayTimeModel1.getSelectTime().add(timeModel);
                        dayTimeRepeatAdapter.notifyDataSetChanged();
                    }
                });
        rvDayTime.setAdapter(dayTimeRepeatAdapter);
        rvDayTime.setLayoutManager(new LinearLayoutManager(this));


        DescriptionRepeatAdapter  adapter = new DescriptionRepeatAdapter(this,image,TraderActivity.this,returnItemView);

        rvDesc.setLayoutManager(new LinearLayoutManager(this));
        rvDesc.setAdapter(adapter);
    }

  /*  private String getCurrentTime() {
        String currentTime="Current Time: "+timepicker.getCurrentHour()+":"+timepicker.getCurrentMinute();
        return currentTime;
    }*/

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {

    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {

    }

    @Override
    public void selectedImage(Bitmap var1, String var2) {

       // ivImg.setImageBitmap(var1);
        if (   imgtype.equals("0"))
        {
            ivImg.setImageBitmap(var1);
        }
        else if (imgtype.equals("1"))
        {
         ivplus.setImageBitmap(var1);
        }
        else if (imgtype.equals("2"))
        {
            ivImg1.setImageBitmap(var1);
        }else if (imgtype.equals("3"))
        {
            ivImg2.setImageBitmap(var1);
        }else if (imgtype.equals("4"))
        {
            ivImg3.setImageBitmap(var1);
        }
        else
        {
            image.put(imasgezpos.toString(),var1);
            Log.e("kmdkmdkedcmk","Activity-"+imasgezpos+"   "+var1);
            Log.e("kmdkmdkedcmk","PPPctivity-"+image.get(imasgezpos));
            DescriptionRepeatAdapter adapter = new DescriptionRepeatAdapter(this,image,TraderActivity.this,returnItemView);
            rvDesc.setLayoutManager(new LinearLayoutManager(this));
            rvDesc.setAdapter(adapter) ;
        }
    }
    public void   imageClick(int pos,int size) {
        imgtype="5";
        returnItemView=size;
        Log.e("wdwdwdd",String.valueOf(pos));
        imasgezpos= String.valueOf(pos);
        image("all");
    }
}

File: app/src/main/java/com/nelyan/ui/TraderActivity.kt
package com.nelyan.ui

import android.content.Context
import android.content.Intent
import android.graphics.Bitmap
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.*
import android.widget.AdapterView.OnItemSelectedListener
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.nelyan.HELPER.image
import com.nelyan.R
import com.nelyan.adapter.DayTimeRepeatAdapter
import com.nelyan.adapter.DayTimeRepeatAdapter.DayTimeRepeatListener
import com.nelyan.adapter.DescriptionRepeatAdapter
import com.nelyan.modals.DayTimeModel
import com.nelyan.modals.TimeModel
import ru.tinkoff.scrollingpagerindicator.ScrollingPagerIndicator
import java.util.*

class TraderActivity : image(), OnItemSelectedListener {
    var mContext: Context? = null
    var ivBack: ImageView? = null
    var ivImg: ImageView? = null
    var ivCam: ImageView? = null
    var btnSubmit: Button? = null
    var tvAdd: TextView? = null
    var tvClock: TextView? = null
    var edClo: TextView? = null
    var edClo1: TextView? = null
    var edClo2: TextView? = null
    var edClo3: TextView? = null
    var ivplus: ImageView? = null
    var ivImg1: ImageView? = null
    var ivImg2: ImageView? = null
    var ivImg3: ImageView? = null
    var ll_1: LinearLayout? = null
    var ll_2: LinearLayout? = null
    var ll_3: LinearLayout? = null
    var orderby: Spinner? = null
    var orderby1: Spinner? = null
    var Recycler_scroll: RecyclerView? = null
    var indicator: ScrollingPagerIndicator? = null
    var rlAddImg: RelativeLayout? = null
    var rlImg: RelativeLayout? = null
    var rvDayTime: RecyclerView? = null
    var rvDesc: RecyclerView? = null
    var dayTimeRepeatAdapter: DayTimeRepeatAdapter? = null
    var returnItemView = 1
    var image = HashMap<String, Bitmap>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_trader)
        mContext = this
        // indicator=findViewById(R.id.indicator);
        /* Recycler_scroll=findViewById(R.id.Recycler_scroll);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getApplicationContext(), LinearLayoutManager.HORIZONTAL, false);
        Recycler_scroll.setLayoutManager(linearLayoutManager);
        ItemsAdapter  adapterItems = new ItemsAdapter(mContext);
        Recycler_scroll.setAdapter(adapterItems);
        indicator.attachToRecyclerView(Recycler_scroll);*/rvDesc = findViewById(R.id.rvDesc)
        rvDayTime = findViewById(R.id.rvDayTime)
        orderby = findViewById(R.id.orderby)
        val country: MutableList<String?> = ArrayList()
        country.add("")
        country.add("USA")
        country.add("Japan")
        country.add("India")
        val arrayAdapte1: ArrayAdapter<*> = ArrayAdapter<Any?>(this, R.layout.customspinner, country as List<Any?>)
        orderby!!.setAdapter(arrayAdapte1)
        ivBack = findViewById(R.id.ivBack)
        btnSubmit = findViewById(R.id.btnSubmit)
        ivBack!!.setOnClickListener(View.OnClickListener { finish() })
        btnSubmit = findViewById(R.id.btnSubmit)
        btnSubmit!!.setOnClickListener(View.OnClickListener {
            val i = Intent(this@TraderActivity, com.nelyan.ui.TraderPreviewActivity::class.java)
            //  i.putExtra("nurActivity","nurFrag");
            startActivity(i)
        })
        ll_1 = findViewById(R.id.ll_1)
        ll_2 = findViewById(R.id.ll_2)
        ll_3 = findViewById(R.id.ll_3)
        ivImg = findViewById(R.id.ivImg)
        ivImg1 = findViewById(R.id.ivImg1)
        ivImg2 = findViewById(R.id.ivImg2)
        ivImg3 = findViewById(R.id.ivImg3)
        rlImg = findViewById(R.id.rlImg)
        rlImg!!.setOnClickListener(View.OnClickListener {
            imgtype = "0"
            image("all")
        })
        ivplus = findViewById(R.id.ivplus)
        rlAddImg = findViewById(R.id.rlAddImg)
        rlAddImg!!.setOnClickListener(View.OnClickListener {
            imgtype = "1"
            image("all")
        })
        ll_1!!.setOnClickListener(View.OnClickListener {
            imgtype = "2"
            image("all")
        })
        ll_2!!.setOnClickListener(View.OnClickListener {
            imgtype = "3"
            image("all")
        })
        ll_3!!.setOnClickListener(View.OnClickListener {
            imgtype = "4"
            image("all")
        })
        val arrayList: ArrayList<TimeModel?> = ArrayList()

        arrayList.add(TimeModel("",""))
        val dayTimeModel = DayTimeModel(arrayList)
        val dayTimeModelArrayList: ArrayList<DayTimeModel> = ArrayList<DayTimeModel>()
        dayTimeModelArrayList.add(dayTimeModel)
        dayTimeRepeatAdapter = DayTimeRepeatAdapter(this, dayTimeModelArrayList,
                object : DayTimeRepeatListener {
                    override fun dayTimeAdd(pos: Int) {
                        val arrayList1: ArrayList<TimeModel?> = ArrayList()

                        arrayList1.add(TimeModel("",""))
                        val dayTimeModel1 = DayTimeModel(arrayList1)
                        dayTimeModelArrayList.add(dayTimeModel1)
                        dayTimeRepeatAdapter!!.notifyDataSetChanged()
                    }

                    override fun timeAdd(pos: Int) {
                        val dayTimeModel1: DayTimeModel = dayTimeModelArrayList[pos]
                        dayTimeModel1.selectTime.add(TimeModel("",""))
                        dayTimeRepeatAdapter!!.notifyDataSetChanged()
                    }
                })
        rvDayTime!!.setAdapter(dayTimeRepeatAdapter)
        rvDayTime!!.setLayoutManager(LinearLayoutManager(this))
        val adapter = DescriptionRepeatAdapter(this, image, this@TraderActivity, returnItemView)
        rvDesc!!.setLayoutManager(LinearLayoutManager(this))
        rvDesc!!.setAdapter(adapter)
    }

    /*  private String getCurrentTime() {
        String currentTime="Current Time: "+timepicker.getCurrentHour()+":"+timepicker.getCurrentMinute();
        return currentTime;
    }*/
    override fun onItemSelected(parent: AdapterView<*>?, view: View, position: Int, id: Long) {}
    override fun onNothingSelected(parent: AdapterView<*>?) {}
    override fun selectedImage(var1: Bitmap, var2: String) {

        // ivImg.setImageBitmap(var1);
        if (imgtype == "0") {
            ivImg!!.setImageBitmap(var1)
        } else if (imgtype == "1") {
            ivplus!!.setImageBitmap(var1)
        } else if (imgtype == "2") {
            ivImg1!!.setImageBitmap(var1)
        } else if (imgtype == "3") {
            ivImg2!!.setImageBitmap(var1)
        } else if (imgtype == "4") {
            ivImg3!!.setImageBitmap(var1)
        } else {
            image[imasgezpos.toString()] = var1
            Log.e("kmdkmdkedcmk", "Activity-$imasgezpos   $var1")
            Log.e("kmdkmdkedcmk", "PPPctivity-" + image[imasgezpos])
            val adapter = DescriptionRepeatAdapter(this, image, this@TraderActivity, returnItemView)
            rvDesc!!.layoutManager = LinearLayoutManager(this)
            rvDesc!!.adapter = adapter
        }
    }

    fun imageClick(pos: Int, size: Int) {
        imgtype = "5"
        returnItemView = size
        Log.e("wdwdwdd", pos.toString())
        imasgezpos = pos.toString()
        image("all")
    }

    companion object {
        var imgtype: String? = null
        var imasgezpos: String? = null
    }
}